frontend/src/app/main/ui/hooks/resize.cljs

```diff
@@ -0,0 +1,105 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) UXBOX Labs SL
+
+(ns app.main.ui.hooks.resize
+  (:require
+   [app.common.geom.point :as gpt]
+   [app.common.logging :as log]
+   [app.main.refs :as refs]
+   [app.util.dom :as dom]
+   [app.util.storage :refer [storage]]
+   [rumext.alpha :as mf]))
+
+(log/set-level! :warn)
+
+(def last-resize-type nil)
+
+(defn set-resize-type! [type]
+  (set! last-resize-type type))
+
+(defn use-resize-hook
+  [key initial min-val max-val axis negate? resize-type]
+
+  (let [current-file-id (mf/deref refs/current-file-id)
+        size-state (mf/use-state (or (get-in @storage [::saved-resize current-file-id key]) initial))
+        parent-ref (mf/use-ref nil)
+
+        dragging-ref (mf/use-ref false)
+        start-size-ref (mf/use-ref nil)
+        start-ref (mf/use-ref nil)
+
+        on-pointer-down
+        (fn [event]
+          (dom/capture-pointer event)
+          (mf/set-ref-val! start-size-ref @size-state)
+          (mf/set-ref-val! dragging-ref true)
+          (mf/set-ref-val! start-ref (dom/get-client-position event))
+          (set! last-resize-type resize-type))
+
+        on-lost-pointer-capture
+        (fn [event]
+          (dom/release-pointer event)
+          (mf/set-ref-val! start-size-ref nil)
+          (mf/set-ref-val! dragging-ref false)
+          (mf/set-ref-val! start-ref nil)
+          (set! last-resize-type nil))
+
+        on-mouse-move
+        (fn [event]
+          (when (mf/ref-val dragging-ref)
+            (let [start (mf/ref-val start-ref)
+                  pos (dom/get-client-position event)
+                  delta (-> (gpt/to-vec start pos)
+                            (cond-> negate? gpt/negate)
+                            (get axis))
+                  start-size (mf/ref-val start-size-ref)
+                  new-size (-> (+ start-size delta) (max min-val) (min max-val))]
+              
+              (reset! size-state new-size)
+              (swap! storage assoc-in [::saved-resize current-file-id key] new-size))))]
+    {:on-pointer-down on-pointer-down
+     :on-lost-pointer-capture on-lost-pointer-capture
+     :on-mouse-move on-mouse-move
+     :parent-ref parent-ref
+     :size @size-state}))
+
+(defn use-resize-observer
+  [callback]
+
+  (let [prev-val-ref (mf/use-ref nil)
+        current-observer-ref (mf/use-ref nil)
+
+        node-ref
+        (mf/use-callback
+         (mf/deps callback)
+         (fn [node]
+           (let [current-observer (mf/ref-val current-observer-ref)
+                 prev-val         (mf/ref-val prev-val-ref)]
+
+             (when (and (not= prev-val node) (some? current-observer))
+               (log/debug :action "disconnect" :js/prev-val prev-val :js/node node)
+               (.disconnect current-observer)
```

Comment:
Remember to use `^js` hint on interop calls on unknown objects, in this case current-observer is clearly unknown to the compiler, and this can cause unexpected issues in advanced compilations

Reply:
The same applies to all interop calls that has unknown type for the compiler.

Reply:
ok, good catch